From 3a3ff137334e9a09f1a20f0d3e0b52fa8967c226 Mon Sep 17 00:00:00 2001
From: Yannick Fertre <yannick.fertre@foss.st.com>
Date: Thu, 1 Feb 2024 09:14:59 +0100
Subject: [PATCH 1035/1695] dts: stm32: add ltdc support on stm32mp2

Add ltdc support on stm32mp2 platform.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 1c8f60f998c25457f30ac8a0e3eb7542980ed25f dts: stm32: add ltdc support on stm32mp23
d3df15f191b845fbe5cfc5860b30895cf0da6bcd dts: stm32: add ltdc support on stm32mp21
ca7112deee09e465d1e36b16abb2a2a0a4c47e57 dts: stm32: add ltdc support on stm32mp2]
Signed-off-by: Yannick Fertre <yannick.fertre@foss.st.com>
Change-Id: Iad07be8198a1ec9549815e07e9a5f7daf3310dd8
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/391787
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
---
 core/arch/arm/dts/stm32mp251.dtsi | 26 ++++++++++++++++++++++++++
 1 file changed, 26 insertions(+)

diff --git a/core/arch/arm/dts/stm32mp251.dtsi b/core/arch/arm/dts/stm32mp251.dtsi
index a7921e179..038d059c4 100644
--- a/core/arch/arm/dts/stm32mp251.dtsi
+++ b/core/arch/arm/dts/stm32mp251.dtsi
@@ -648,6 +648,32 @@
 					status = "disabled";
 				};
 			};
+
+			ltdc: display-controller@48010000 {
+				compatible = "st,stm32-ltdc";
+				reg = <0x48010000 0x400>;
+				interrupts = <GIC_SPI 160 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 161 IRQ_TYPE_LEVEL_HIGH>;
+				clocks = <&rcc CK_BUS_LTDC>, <&rcc CK_KER_LTDC>;
+				clock-names = "bus", "lcd";
+				resets = <&rcc LTDC_R>;
+				status = "disabled";
+				access-controllers = <&rifsc STM32MP25_RIFSC_LTDC_CMN_ID>;
+				access-controller-names = "cmn";
+
+				l1l2 {
+					access-controllers = <&rifsc STM32MP25_RIFSC_LTDC_L0L1_ID>;
+					access-controller-names = "l1l2";
+				};
+				l3 {
+					access-controllers = <&rifsc STM32MP25_RIFSC_LTDC_L2_ID>;
+					access-controller-names = "l3";
+				};
+				rot {
+					access-controllers = <&rifsc STM32MP25_RIFSC_LTDC_ROT_ID>;
+					access-controller-names = "rot";
+				};
+			};
 		};
 
 		iac: iac@42090000 {
-- 
2.47.2

